### ISSUE TITLE
AED + new empire announcement event + nomads = event spam

### ISSUE THREAD

**Issue by zijistark** [zijistark/EMF#574]
This is a classic issue, but it does need to be solved if possible. I don't actually remember the specifics of the causal chain of events which is causing this, but it's time to remember. It seems a nomad is settling as tribe and the engine is auto-generating an empire title for him at that time; his realm is not big enough for AED's requirements, so AED destroys his tribal empire; for some reason, doing this then turns him back into nomad (?); and around and around we go.

![image](https://user-images.githubusercontent.com/6268890/52287158-83e39a80-291e-11e9-9140-b6376fcfb7d7.png)

````
[HIP Release 2019-02-04 (Furry1 Beta3)]
EMF: Extended Mechanics & Flavor (v9.0-BETA - 2019-02-05)
SWMH (v3.2-BETA - 2019-02-01)
SED: English Localisation for SWMH (v2.2.19 - 2019-02-05)
ARKO Armoiries (2019-02-03)
LTM (C - 2012-02-03)
CPRplus (v3.0.1 - 2019-01-02)
````

User said:
> My game is getting spammed every week because this nomad keeps founding and ending his empire on the same day over and over

**Comment by zijistark** [comment]
It turns out that this is actually a much bigger issue than I thought, partially worsened by the decision Rylock added for altaic rulers to convert back to being nomads.  Either way, what we get is the occasional occurrence of hard-looping between founding an empire and the destruction of the empire, which if it does happen in your game, is somewhat of a showstopper by our standards.

**Comment by zijistark** [comment]
Another case:
![image](https://user-images.githubusercontent.com/6268890/52385195-6f45f600-2a35-11e9-8cf3-3f773f2f6f86.png)

This is really not making any sense to me. I coded up what I thought would fix this, and I tried to reproduce by taking away all blockers (time or otherwise) to nomad settlement and AED.  I didn't encounter the issue when doing this (with the new code albeit).  However, if you inspect what's going on in these screenshots, it seems that I was trying to fix an entirely different problem. I'm a bit stumped.

**Comment by zijistark** [comment]
Closing for now, as we've had no reports of the issue after I took some experimental measures to stop it. Can reopen if it's still an issue.

**Comment by zijistark** [comment]
It looks like the changes I've made over various commits to the `emf_crier` module's "Founding of a New Empire" announcement event has stopped that part of the problem with nomads. However, the "End of an Empire" announcements are still spamming for some users, so clearly the measures I took to auto-destroy the hard-code-generated empire titles when they change government to non-nomadic were not sufficient.

I propose writing a scripted effect wrapper for every type of government type to which a nomad can switch (this is more than just tribal, because nomads can be adventurers too, and adventurer governments get switched to be appropriate for what they've conquered once landed. Additionally, the vanilla *Nomad Agitation* mechanic gets up to similar shenanigans.

In this scripted trigger template (for which the only thing that varies is the target government type in a `set_government_type = <government type tag>` command within it), we'd check if the government-switcher was nomadic currently, and if so, flag all of their non-nomadic titles right before the government switch, and right afterward, destroy any EMPEROR-tier titles that weren't flagged (and then clear all the flags), as they must have been hard-code-generated by the government switch.
